### PR TITLE
Add hatch-test env customization

### PR DESCRIPTION
**Pull request by flying-sheep**
This allows us to locally test with an approximation of our online test envs:

```console
$ hatch test # regular, stable resolution. Also `hatch test -i deps=stable`
$ hatch test -i deps=pre  # pre-releases
$ hatch test -i deps=min  # Python 3.0 and minimum possible dep versions.
$ hatch test --all
```

Approximation because `UV_RESOLUTION=lowest-direct` doesn’t directly match our min-deps ci script. Maybe we should get rid of the custom version number semantics in that one and just switch to `UV_RESOLUTION=lowest-direct`

---

I want this because I had a custom `hatch.toml` that got nuked when we added one to the branch, oops!

### CODE REVIEW

File: hatch.toml [hunk 1]
```diff
@@ -9,3 +9,19 @@ dependencies = ["setuptools"]  # https://bitbucket.org/pybtex-devs/pybtex/issues
 [envs.docs.scripts]
 build = "sphinx-build -M html docs docs/_build -W --keep-going {args}"
 clean = "git clean -fX -- docs"
+
+[envs.hatch-test]
+default-args = []
+extra-dependencies = ["ipykernel"]
```
Comment (ilan-gold) [inline hatch.toml:15]: Why `ipykernel`?

File: hatch.toml [hunk 2]
```diff
@@ -9,3 +9,19 @@ dependencies = ["setuptools"]  # https://bitbucket.org/pybtex-devs/pybtex/issues
 [envs.docs.scripts]
 build = "sphinx-build -M html docs docs/_build -W --keep-going {args}"
 clean = "git clean -fX -- docs"
+
+[envs.hatch-test]
```
Comment (ilan-gold) [inline hatch.toml:13]: Why not use `uv` by default? Is this inherited from `default`?

File: hatch.toml (hunk 1 above)
Comment (flying-sheep) [reply hatch.toml:15]: Allows you to use this env as kernel for notebooks in VS Code.

File: hatch.toml (hunk 2 above)
Comment (flying-sheep) [reply hatch.toml:13]: - all envs *except* the `hatch-*` ones inherit from default, that’s why this works without `envs.towncrier.features` being defined: https://github.com/scverse/anndata/blob/94b2304cb2c08dd2c5f8d3cb7ee4da39d874975a/hatch.toml#L13-L15
- the `hatch-*` envs have `uv` by default

**Review by ilan-gold — APPROVED** [review]

**Pull request merged**